hudi-spark-datasource/hudi-spark3.2plus-common/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -140,7 +157,11 @@ class HoodieCatalog extends DelegatingCatalogExtension
                            properties: util.Map[String, String]): Table = {
     if (sparkAdapter.isHoodieTable(properties)) {
       val locUriAndTableType = deduceTableLocationURIAndTableType(ident, properties)
-      createHoodieTable(ident, schema, locUriAndTableType, partitions, properties,
+      val partitionTransforms =
+        if (partitions.isEmpty && isTablePartitioned(getExistingTableIfExists(ident.asTableIdentifier)))
+          buildPartitionTransforms(spark, locUriAndTableType._1.getPath)
+        else partitions
```

Comment:
nit: this common logic can be extracted to a util method

Reply:
Sure, I'll refactor it in a subsequent PR.